invenio_rdm_records/services/access/service.py

```diff
@@ -749,3 +756,41 @@ def create_guest_access_request(self, identity, token, expand=False, uow=None):
             data=comment,
             uow=uow,
         )
+
+    @unit_of_work()
+    def update_access_settings(
+        self,
+        identity,
+        id_,
+        data,
+        uow=None,
+    ):
+        """Update access settings for a record (resp. its parent)."""
+        record, parent = self.get_parent_and_record_or_draft(id_)
+
+        # Permissions
+        self.require_permission(identity, "manage", record=record)
+
+        # Validation
+        data, __ = self.schema_parent_access_settings.load(
+            data, context=dict(identity=identity), raise_errors=True
+        )
+
+        # Update
+        for key in data:
+            setattr(parent.access.settings, key, data[key])
```

Comment:
could't we do simply 
```suggestion
            setattr(parent.access, "settings", data)
```